train.py

```diff
@@ -452,11 +462,22 @@ def train(hyp, opt, device, tb_writer=None):
     # Train
     logger.info(opt)
     if not opt.evolve:
-        tb_writer = None
+        tb_writer, wandb = None, None  # init loggers
         if opt.global_rank in [-1, 0]:
+            # Tensorboard
             logger.info(f'Start Tensorboard with "tensorboard --logdir {opt.logdir}", view at http://localhost:6006/')
             tb_writer = SummaryWriter(log_dir=log_dir)  # runs/exp0
 
+            # W&B
+            try:
+                import wandb
+
+                assert os.environ.get('WANDB_DISABLED') != 'true'
+                logger.info("Weights & Biases logging enabled, to disable set os.environ['WANDB_DISABLED'] = 'true'")
+            except (ImportError, AssertionError):
+                opt.log_imgs = 0
+                logger.info("Install Weights & Biases for experiment logging via 'pip install wandb' (recommended)")
+
         train(hyp, opt, device, tb_writer)
```

Comment:
@glenn-jocher looks like you forgot to pass `wandb` argument here. Logging will not be enabled even if wandb is installed if this is the case. Maybe set `wandb=True` when the import is successful and pass it in the train function

Reply:
Ah of course, you are right! Yes this needs to be passed, and then inside train.py it is used normally if a package, or ignored if None.

Reply:
Ok, I've just pushed a bug fix commit that should resolve this.